eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs

```diff
@@ -168,17 +335,58 @@ pattern TxOut ::
   StrictMaybe (DataHash (Crypto era)) ->
   TxOut era
 pattern TxOut addr vl dh <-
-  (viewCompactTxOut -> (addr, vl, dh))
+  (viewTxOut -> (addr, vl, dh))
   where
+    TxOut (Addr network paymentCred stakeRef) vl SNothing
+      | adaOnly vl,
+        Just Refl <- sameNat (Proxy @(SizeHash (CC.ADDRHASH (Crypto era)))) (Proxy @28) =
+        let (a, b, c, d) = encodeAddress28 network paymentCred
+         in TxOut_AddrHash28_AdaOnly stakeRef a b c d (fromIntegral (unCoin (coin vl)))
+    TxOut (Addr network paymentCred stakeRef) vl (SJust dh)
+      | adaOnly vl,
+        Just Refl <- sameNat (Proxy @(SizeHash (CC.ADDRHASH (Crypto era)))) (Proxy @28),
+        Just Refl <- sameNat (Proxy @(SizeHash (CC.HASH (Crypto era)))) (Proxy @32) =
+        let (a, b, c, d) = encodeAddress28 network paymentCred
+            (e, f, g, h) = encodeDataHash32 dh
+         in TxOut_AddrHash28_AdaOnly_DataHash32 stakeRef a b c d (fromIntegral (unCoin (coin vl))) e f g h
     TxOut addr vl mdh =
       let v = fromMaybe (error $ "Illegal value in txout: " <> show vl) $ toCompact vl
           a = compactAddr addr
        in case mdh of
-            SNothing -> TxOutCompact a v
-            SJust dh -> TxOutCompactDH a v dh
+            SNothing -> TxOutCompact' a v
+            SJust dh -> TxOutCompactDH' a v dh
 
 {-# COMPLETE TxOut #-}
 
+-- TODO deprecate
+pattern TxOutCompact ::
+  ( Era era,
+    HasCallStack
+  ) =>
+  CompactAddr (Crypto era) ->
+  CompactForm (Core.Value era) ->
+  TxOut era
+pattern TxOutCompact addr vl <-
+  (viewCompactTxOut -> (addr, vl, SNothing))
+  where
+    TxOutCompact = TxOutCompact'
+
+-- TODO deprecate
```

Comment:
I think it makes sense to add `DEPRECATED` pragmas right away.


```suggestion
{-# DEPRECATED TxOutCompactDH "In favor of `TxOut`. Do not use compacted versions directly." #-}
```

Reply:
So these patterns are still used in this repo which causes warnings. Do you know if we allow warning in the codebase? I was aiming to simply do a follow up PR removing this. Would that be sufficient?

Reply:
Yeah, you are right. Disregard my comment, we'll deal with this at some later point.

Reply:
These patterns are actually completely wrong, because when using them the new compact form cannot be constructed. Therefore in all places where `TxOutCompact` and `TxOutCompactDH` are used for constructing `TxOut` will always construct the less optimal `TxOutCompact'` and `TxOutCompactDH'`